numpyro/distributions/mixtures.py

```diff
@@ -320,11 +319,9 @@ def __init__(
         # TODO: It would be good to check that the support of all the component
         # distributions match, but for now we just check the type, since __eq__
         # isn't consistently implemented for all support types.
-        support_type = type(component_distributions[0].support)
-        if any(
-            type(d.support) is not support_type for d in component_distributions[1:]
-        ):
-            raise ValueError("All component distributions must have the same support.")
+        # support_type = type(component_distributions[0].support)
```

Comment:
You can introduce a `support` argument:
```
def __init__(
        self, mixing_distribution, component_distributions, *, support=None, validate_args=None
    ):
```
and we will execute this check when `support=None`.

Reply:
I might be missing something, but I think this check is unnecessary. Can you help me understand why we need to check the types again, even after we've already resolved the issue for different support scenarios?

Reply:
If supports are different, we don't have a mechanism to obtain support of the mixture. We currently use the support of the first component, which only works when all supports are the same.

Reply:
Seems reasonable, I am adding this change too.